src/client/session/Context.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { createContext, useReducer, ReactNode, useCallback } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { CardProps, InputType, SortDirection } from '../../model/Card'
+import { GameState } from '../../model/Game'
+
+export interface GameResultItem extends CardProps {
+  answer?: string
+  isAccepted?: boolean
+}
+
+type State = {
+  gameCards: GameResultItem[]
+  storeCards: CardProps[]
+  sortType: InputType
+  sortDirection: SortDirection
+  cardInput: { front: string; back: string }
+  filterChecked: boolean
+  gameState: GameState
+  buttonLabel: string
+  currentCardIndex: number
+}
+
+export type Action =
+  | { type: 'INIT_GAME'; payload: GameResultItem[] }
+  | { type: 'SET_CARD_INDEX'; payload: number }
+  | { type: 'FINISH_GAME' }
+  | { type: 'DELETE_GAME' }
+  | { type: 'SET_CARDS'; payload: GameResultItem[] }
+  | { type: 'ADD_CARD'; payload: CardProps }
+  | { type: 'DELETE_CARD'; payload: string }
+  | { type: 'SET_SORT_TYPE'; payload: InputType }
+  | { type: 'SET_SORT_DIRECTION'; payload: SortDirection }
+  | { type: 'SET_CARD_INPUT'; payload: { front: string; back: string } }
+  | { type: 'SET_FILTER_CHECKED'; payload: boolean }
```

Comment:
Ich finde diese Schreibweise eigentlich ausreichend. Glaubst du, dass tobi die typisierte Variante bevorzugt. Mir kam das in dem Falle so vor... Glaube aber das es nur Stiel ist und nicht unbedingt bewertbar.

A: Wir lassens so :+1: 
B: Wir typisieren und lagern es in ein eigenes File aus mit Actions (Wie in seinen Beispielen...) :-1: 

Viel mehr interessiert mich aber, ob das deine bevorzugte Variante ist

Reply:
Also rein von der Typensicherheit macht es keinen Unterschied und da wir die Actions nicht modular aufbauen gibt es in der aktuellen Ausführung keinen wirklichen Nachteil. Ich schlage vor wir lassen das so, weil einfacher lesbar? Können sonst beim Refactoring vor Abgabe noch kurz schauen, ob wir es auslagern wollen 😉